src/sentry/tasks/assemble.py

```diff
@@ -1,29 +1,100 @@
 from __future__ import absolute_import, print_function
 
+from os import path
+
+import hashlib
 import logging
+import six
+
+from django.db import IntegrityError, transaction
 
 from sentry.api.serializers import serialize
+from sentry.cache import default_cache
 from sentry.tasks.base import instrumented_task
+from sentry.utils import json
 from sentry.utils.files import get_max_file_size
 from sentry.utils.sdk import configure_scope
 
 logger = logging.getLogger(__name__)
 
 
+def enum(**named_values):
+    """Creates an enum type."""
+    return type('Enum', (), named_values)
+
+
+ChunkFileState = enum(
+    OK='ok',  # File in database
+    NOT_FOUND='not_found',  # File not found in database
+    CREATED='created',  # File was created in the request and send to the worker for assembling
+    ASSEMBLING='assembling',  # File still being processed by worker
+    ERROR='error'  # Error happened during assembling
+)
+
+
+AssembleTask = enum(
+    DIF='project.dsym',  # Debug file upload
+    ARTIFACTS='organization.artifacts',  # Release file upload
+)
+
+
+def _get_cache_key(task, scope, checksum):
+    """Computes the cache key for assemble status.
+
+    ``task`` must be one of the ``AssembleTask`` values. Scope can be any model
+    that has an identifier, such as the organization or project that this task
+    is performed under.
+
+    ``checksum`` should be the SHA1 hash of the main file that is being
+    assembled.
+    """
+    return 'assemble-status:%s' % hashlib.sha1(b'%s|%s|%s' % (
+        str(scope.id).encode('ascii'),
```

Comment:
Would prefer not to rely on `scope.id` instead, pass in the id directly.
It's not clear from the param name that the object passed in there should have an `id` prop.